### PR TITLE
fixing NRP networking resource properties

### PR DESCRIPTION
**Pull request by guptas14**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by kairu-ms** [comment]
@guptas14 Please attach s360 issue links. Thanks.

**Comment by kairu-ms** [comment]
Hi @guptas14, Please ping supports to fix CI issues. Thanks.

**Comment by guptas14** [comment]
S360 identified issues link:
https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8187877

**Comment by guptas14** [comment]
> Hi @guptas14, Please ping supports to fix CI issues. Thanks.

Hi @kairu-ms , Could you elaborate here, whom should I ping? Sorry, I didn't quite get you.

**Review by kairu-ms — COMMENTED** [review]
Please update related examples.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json [hunk 1]
```diff
@@ -83,7 +83,7 @@
     "CloudError": {
       "x-ms-external": true,
       "properties": {
-        "error": {
+        "Error": {
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json:86]: @msyyc @jsntcy Please review this change. Will it impact CLI?

**Comment by msyyc** [comment]
According to codegen log, there may be problems in swagger file. Detailed info : 

cmderr	[Autorest] Failure during batch task - {"tag":"package-2020-08"} -- Error: $ref to original location 'file:///home/vsts/work/1/s/azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json#/components/schemas/PrivateLinkService' is not found in the new refs collection.
cmderr	[Autorest]   Error: $ref to original location 'file:///home/vsts/work/1/s/azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json#/components/schemas/PrivateLinkService' is not found in the new refs collection
error	Script return with result [failed] code [1] signal [null] cwd [azure-sdk-for-python]: autorest
warn	Skip package processing as generation is failed

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json (hunk 1 above)
Comment (jsntcy) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json:86]: @guptas14, Could you share the background why you need change `error` to `Error`?

**Comment by jsntcy** [comment]
@guptas14, please share the background why you did these changes.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json (hunk 1 above)
Comment (guptas14) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json:86]: Hi,
I need to make these changes to fix the Swagger incorrectness identified by S360 automation.
There are several resource properties being returned by Networking RP service, which were missing in Swagger, or which did not tell default values.
For Error property, the Swagger definition has 'e' where all api's return error with 'E' in json object. This CloudError is a generic object for all error properties being returned.



S360 identified issues link:
https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8187877

Comment (guptas14) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/network.json:86]: Please check this report, for instance, generated by S360:
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=136094B8-8B18-40F7-A370-07AC2CCA81B0

Additional properties not allowed: Error
Json path: $.Error
#/definitions/CloudError
network.json#L83

@shenglol Could you please confirm here, regarding these Swagger incorrectness fixes required by NRP.

**Comment by guptas14** [comment]
> According to codegen log, there may be problems in swagger file. Detailed info :
> 
> cmderr [Autorest] Failure during batch task - {"tag":"package-2020-08"} -- Error: $ref to original location 'file:///home/vsts/work/1/s/azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json#/components/schemas/PrivateLinkService' is not found in the new refs collection.
> cmderr [Autorest] Error: $ref to original location 'file:///home/vsts/work/1/s/azure-rest-api-specs/specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json#/components/schemas/PrivateLinkService' is not found in the new refs collection
> error Script return with result [failed] code [1] signal [null] cwd [azure-sdk-for-python]: autorest
> warn Skip package processing as generation is failed

Could you help me with the fix needed here?

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json [hunk 2]
```diff
@@ -444,6 +444,18 @@
           "readOnly": true,
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the public IP address resource."
+        },
+        "servicePublicIPAddress": {
+          "$ref": "#/definitions/PublicIPAddress",
+          "description": "The service public IP address of the public IP address resource."
+        },
+        "natGateway": {
+          "$ref": "./network.json#/definitions/NatGateway",
```
Comment (raych1) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json:453]: This reference model doesn't exist.

Comment (guptas14) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json:453]: @raych1  Could you guide how can I mention the reference to NatGateway model here? what should be the correct syntax?

Comment (raych1) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json:453]: The syntax is correct and the issue is NatGateway doesn't exist in network.json under 2020-08-01 folder.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpPrefix.json [hunk 3]
```diff
@@ -437,6 +437,10 @@
           "readOnly": true,
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the public IP prefix resource."
+        },
+        "natGateway": {
+          "$ref": "./network.json#/definitions/NatGateway",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpPrefix.json:442]: I think the NatGateway is defined in netGateway.json not in network.json

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpPrefix.json [hunk 4]
```diff
@@ -437,6 +437,10 @@
           "readOnly": true,
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the public IP prefix resource."
+        },
+        "natGateway": {
+          "$ref": "./network.json#/definitions/NatGateway",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpPrefix.json:442]: I think the NatGateway is defined in netGateway.json not in network.json

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpPrefix.json (hunk 3 above)
Comment (guptas14) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpPrefix.json]: resolved

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpPrefix.json (hunk 4 above)
Comment (guptas14) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpPrefix.json]: resolved

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/networkInterface.json [hunk 5]
```diff
@@ -1153,6 +1157,14 @@
           "readOnly": true,
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the network interface resource."
+        },
+        "nicType": {
+          "type": "string",
+          "description": "Type of Network Interface resource."
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/networkInterface.json:1163]: Is it an enumable property? If so, you should set it in enum type.

Comment (kairu-ms) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/networkInterface.json:1163]: https://github.com/Azure/azure-rest-api-specs/blob/5015df257c6f0f8f55e2491a5c13b6ac589c5288/specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/customIpPrefix.json#L366-L381

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpAddress.json [hunk 6]
```diff
@@ -444,6 +444,18 @@
           "readOnly": true,
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the public IP address resource."
+        },
+        "natGateway": {
+          "$ref": "./natGateway.json#/definitions/NatGateway",
+          "description": "The NatGateway for the Public IP address."
+        },
+        "migrationPhase": {
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpAddress.json:452]: Is it an enumable property? If so, you should set it in enum type.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/serviceTags.json [hunk 7]
```diff
@@ -163,6 +163,11 @@
           },
           "readOnly": true,
           "description": "The list of IP address prefixes."
+        },
+        "state": {
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/serviceTags.json:167]: Is it an enumable property? If so, you should set it in enum type.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json [hunk 8]
```diff
@@ -1496,10 +1496,12 @@
         },
         "privateEndpointNetworkPolicies": {
           "type": "string",
+          "default": "Enabled",
           "description": "Enable or Disable apply network policies on private end point in the subnet."
         },
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json:1501]: You should write it in enum format like this:

https://github.com/Azure/azure-rest-api-specs/blob/5015df257c6f0f8f55e2491a5c13b6ac589c5288/specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/applicationGateway.json#L1786-L1797

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json [hunk 9]
```diff
@@ -1496,10 +1496,12 @@
...
         "privateLinkServiceNetworkPolicies": {
           "type": "string",
+          "default": "Enabled",
           "description": "Enable or Disable apply network policies on private link service in the subnet."
         }
       },
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json:1507]: https://github.com/Azure/azure-rest-api-specs/blob/5015df257c6f0f8f55e2491a5c13b6ac589c5288/specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/applicationGateway.json#L1786-L1797

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json [hunk 10]
```diff
@@ -1153,6 +1157,19 @@
           "readOnly": true,
           "$ref": "./network.json#/definitions/ProvisioningState",
           "description": "The provisioning state of the network interface resource."
+        },
+        "nicType": {
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json:1161]: Is it an enumable property? If so, you should set it in enum type.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json [hunk 11]
```diff
@@ -1153,6 +1157,19 @@
...
+          "default": "Standard",
+          "description": "Type of Network Interface resource."
+        },
+        "privateLinkService": {
+          "$ref": "./privateLinkService.json#/definitions/PrivateLinkService",
+          "description": "Privatelinkservice of the network interface resource."
+        },
+        "migrationPhase": {
+          "type": "string",
+          "description": "Migration phase of Public IP Address."
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json:1172]: Is it an enumable property? If so, you should set it in enum type.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json [hunk 12]
```diff
@@ -444,6 +444,22 @@
...
+          "description": "The service public IP address of the public IP address resource."
+        },
+        "natGateway": {
+          "$ref": "./natGateway.json#/definitions/NatGateway",
+          "description": "The NatGateway for the Public IP address."
+        },
+        "migrationPhase": {
+          "type": "string",
+          "description": "Migration phase of Public IP Address."
+        },
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json:459]: Is it an enumable property? If so, you should set it in enum type.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/virtualNetwork.json [hunk 13]
```diff
@@ -1496,11 +1500,20 @@
         },
         "privateEndpointNetworkPolicies": {
           "type": "string",
+          "default": "Enabled",
           "description": "Enable or Disable apply network policies on private end point in the subnet."
         },
         "privateLinkServiceNetworkPolicies": {
           "type": "string",
+          "default": "Enabled",
           "description": "Enable or Disable apply network policies on private link service in the subnet."
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/virtualNetwork.json:1509]: You should write it in enum format like this:

https://github.com/Azure/azure-rest-api-specs/blob/5015df257c6f0f8f55e2491a5c13b6ac589c5288/specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/applicationGateway.json#L1786-L1797

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/serviceTags.json (hunk 7 above)
Comment (guptas14) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/serviceTags.json:167]: This is not an enum.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/virtualNetwork.json [hunk 14]
```diff
@@ -1496,11 +1500,36 @@
...
           "type": "string",
-          "description": "Enable or Disable apply network policies on private end point in the subnet."
+          "default": "Enabled",
+          "description": "Enable or Disable apply network policies on private end point in the subnet.",
+          "enum": [ 
+            "Enabled", 
+            "Disabled" 
+          ], 
+          "x-ms-enum": { 
+            "name": "PrivateEndpointNetworkPolicies",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/virtualNetwork.json:1510]: Please add Prefix for enum name, For example "VirtualNetworkPrivateEndpointNetworkPolicies"

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json [hunk 15]
```diff
@@ -1153,6 +1157,38 @@
...
+        "nicType": {
+          "type": "string",
+          "default": "Standard",
+          "description": "Type of Network Interface resource.",
+          "enum": [
+            "Standard",
+            "Elastic"
+          ],
+          "x-ms-enum": { 
+            "name": "NicType",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json:1170]: Please add Prefix for enum name

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json [hunk 16]
```diff
@@ -1153,6 +1157,38 @@
...
+          "description": "Migration phase of Network Interface resource.",
+          "enum": [
+            "None",
+            "Prepare",
+            "Commit",
+            "Abort",
+            "Committed"
+          ],
+          "x-ms-enum": { 
+            "name": "MigrationPhase",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/networkInterface.json:1189]: Please add Prefix for enum name

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json [hunk 17]
```diff
@@ -1496,11 +1496,29 @@
...
           "type": "string",
-          "description": "Enable or Disable apply network policies on private link service in the subnet."
+          "default": "Enabled",
+          "description": "Enable or Disable apply network policies on private link service in the subnet.",
+          "enum": [ 
+            "Enabled", 
+            "Disabled" 
+          ], 
+          "x-ms-enum": { 
+            "name": "PrivateLinkServiceNetworkPolicies",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json:1519]: Please add Prefix for enum name

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json [hunk 18]
```diff
@@ -1496,11 +1496,29 @@
...
           "type": "string",
-          "description": "Enable or Disable apply network policies on private end point in the subnet."
+          "default": "Enabled",
+          "description": "Enable or Disable apply network policies on private end point in the subnet.",
+          "enum": [ 
+            "Enabled", 
+            "Disabled" 
+          ], 
+          "x-ms-enum": { 
+            "name": "PrivateEndpointNetworkPolicies",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/virtualNetwork.json:1506]: Please add Prefix for enum name

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpAddress.json [hunk 19]
```diff
@@ -444,6 +444,29 @@
...
+          "description": "Migration phase of Public IP Address.",
+          "enum": [
+            "None",
+            "Prepare",
+            "Commit",
+            "Abort",
+            "Committed"
+          ],
+          "x-ms-enum": { 
+            "name": "MigrationPhase",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/publicIpAddress.json:463]: Please add Prefix for enum name

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/networkInterface.json [hunk 20]
```diff
@@ -1153,6 +1157,22 @@
...
+        },
+        "nicType": {
+          "type": "string",
+          "description": "Type of Network Interface resource.",
+          "enum": [
+            "Standard",
+            "Elastic"
+          ],
+          "x-ms-enum": {
+            "name": "NicType",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/networkInterface.json:1169]: Need prefix

File: specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json [hunk 21]
```diff
@@ -444,6 +444,33 @@
...
+          "description": "Migration phase of Public IP Address.",
+          "enum": [
+            "None",
+            "Prepare",
+            "Commit",
+            "Abort",
+            "Committed"
+          ],
+          "x-ms-enum": {
+            "name": "MigrationPhase",
```
Comment (kairu-ms) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-08-01/publicIpAddress.json:467]: need prefix

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**